revisions.js

```diff
@@ -68,14 +68,109 @@
+//   { firstName: "Martina", lastName: "Freundorfer" },
+//   { firstName: "Leon-Arno", lastName: "Rimrod" },
+//   { firstName: "Itamar", lastName: "Givon" }
+// ];
 
 /**
  * Exercise:
  * Map the array above into a new array containing just the last names
  */
+// const profiles = [
+//   {firstName: "Robert", lastName: "Rostock"},
+//   {firstName: "Martina", lastName: "Freundorfer"},
+//   {firstName: "Leon-Arno", lastName: "Rimrod"},
+//   {firstName: "Itamar", lastName: "Givon"},
+// ]
+
+// const newProfiles = profiles.map(profile => {
+//   var newArray = profile.lastName;
+//   return newArray
+// }
+// )
+
+// // console.log(newProfiles)
+
+// const newProfiles = profiles.map(profile => {
+//     return profile.lastName;})
+
+//     console.log(newProfiles)
+
+let classmates = [
+  { firstName: "Naima", lastName: "Adan Ahmed", Age: 29 },
+  { firstName: "Balazs", lastName: "Danyadi", Age: 33 },
+  { firstName: "Givara", lastName: "Mahfoud", Age: 25 },
+  { firstName: "Ion", lastName: "Lazarev", Age: 31 },
+  { firstName: "Nic", lastName: "Solitom", Age: 33 },
+  { firstName: "Mojgan", lastName: "Vafa", Age: 45 },
+  { firstName: "Hamida", lastName: "Shinri", Age: 29 },
+  { firstName: "Karol", lastName: "Polakowski", Age: 29 },
+  { firstName: "Christophe", lastName: "Troudart", Age: 29 },
+  { firstName: "Emma", lastName: "Blabla", Age: 29 },
+  { firstName: "Bianca", lastName: "Richa", Age: 33 }
+];
+
+console.log(classmates);
+
+// const newClassmates = classmates.map(classmates => {
+//   return `${classmates.firstName}  ${classmates.lastName} is ${classmates.Age} years old`;})
+
+//   console.log(newClassmates)
+
+//   let sum = 0;
+//   classmates.forEach(classmates => {
+//     sum += classmates.Age;
+//   })
+
+//   console.log(sum)
+
+
+
+// const totalAge = classmates.reduce ((accumulator, object) =>{
+//   return accumulator + object.Age;
+// },0)
+
+// console.log(totalAge)
+
+
+
+
+// Exercise: Reduce the classMates array to a boolean that indicates whether there is an object with the firstName Mojgan is in it.
+
+// We can use a reducer to check if a value is in an array...
+// const wheresMojgan = classMates.reduce((hasMojgan, student) => {
+//   return hasMojgan || student.firstName === "Mojgan";
+// }, false);
+
+// console.log(wheresMojgan);
+
+// ... Buuuut, Array.prototype.includes is much better at it
+const isMojganHere = classmates
+  .map(student => {
+    return student.firstName;
+  })
+  .includes("Mojgan");
+
+console.log(isMojganHere);
+
+// Exercise: Reduce the array of students int a a string of first nam and age seperated by commas. Ex: Nic is 32, Naime is 29...
+// Do this same task in *TWO* different ways
+
+const sentences = classmates.reduce((acc,student)=>{
+  return `${acc} ${student.firstName}  is ${student.Age}.`;
+},"")
+
+console.log(sentences)
+
+
+
+
+const anotherSentence = classmates.map((students)=>{
+  return `${students.firstName} is ${students.Age}.`
+});
```

Comment:
Fantastic, now we're only missing one thing to make it into a single string. I wonder how we can `Array.prototype.join` these strings together... :thinking: